core/source/general/block_array.hpp

```diff
@@ -331,11 +334,11 @@ template <typename IndexType, typename EntryType> class dynarray_block_array :
 			EntryType* block = this->blocks[blockIndex];
 			if (block)
 			{
-				for (IndexType i = 0; i < blockLength; ++i)
+				for (IndexType i = 0; i < this->block_array<IndexType, EntryType>::blockLength; ++i)
```

Comment:
I suspect this can be changed back to this->blockLength with the member removed? I had changed the base class private stuff to protected, so should  be visible in derived class?